src/services/projects.js

```diff
@@ -470,131 +523,334 @@ module.exports = class ProjectsHelper {
 				['id', 'value', 'has_entities', 'validations']
 			)
 
-			// Using forEach instead of for loop for entity type validations
+			//fetch task entityType validations
+			let taskEntityTypes = await entityModelMappingQuery.findEntityTypesAndEntities(
+				{
+					model: common.TASKS,
+					status: common.STATUS_ACTIVE,
+				},
+				userDetails.organization_id,
+				['id', 'value', 'validations', 'has_entities']
+			)
+
+			const taskEntityTypesMapping = taskEntityTypes.reduce((acc, item) => {
+				acc[item.value] = item
+				return acc
+			}, {})
+
+			//get all entity type validation for file types
+			let allowedFileTypesEntityTypes = await entityTypeQueries.findUserEntityTypeAndEntities(
+				{
+					value: common.TASK_ALLOWED_FILE_TYPES,
+					status: common.STATUS_ACTIVE,
+				},
+				userDetails,
+				['id', 'value', 'has_entities', 'validations']
+			)
 
-			entityTypes.forEach((entityType) => {
-				const fieldData = projectData[entityType.value]
+			allowedFileTypesEntityTypes =
+				allowedFileTypesEntityTypes.length > 0 ? allowedFileTypesEntityTypes[0] : allowedFileTypesEntityTypes
+			const allowedFileTypesEntities = allowedFileTypesEntityTypes.entities.map((entity) => {
+				return entity.value
+			})
 
+			for (let entityType of entityTypes) {
+				let fieldData = projectData[entityType.value]
+				//check field is required
 				if (entityType.validations.required) {
 					let required = utils.checkRequired(entityType, fieldData)
 					if (!required) {
 						throw {
-							message: `${entityType.value} not added`,
-							error: utils.errorObject(common.BODY, entityType.value),
+							error: utils.errorObject(common.BODY, entityType.value, `${entityType.value} is required`),
+						}
+					}
+				}
+
+				// check max character limit exceeded
+				if (Object.keys(entityType.validations).includes(common.MAX_CHARACTER_LIMIT) && fieldData) {
```

Comment:
@priyanka-TL Why do we need to do like this, it can all be done via regex - https://www.sitepoint.com/using-regular-expressions-to-check-string-length/

Reply:
Already regex is there for special character check, so we cant send valid messages, hence added one more key in the validation itself.

Reply:
@priyanka-TL As discussed you will add it in regex and not via env